api/types/kustomization.go

```diff
@@ -165,3 +172,20 @@ func (k *Kustomization) EnforceFields() []string {
 	}
 	return errs
 }
+
+// Unmarshal replace k with the content in YAML input y
+func (k *Kustomization) Unmarshal(y []byte) error {
```

Comment:
Please add a kustomization_test.go file.  All public methods need coverage.
Include a complex success case (lots of fields) and a simple failure case.
I realize there's no test file.  The bar needs raising. This file used to be struct only.